src/lib.rs

```diff
@@ -388,6 +425,7 @@ cfg_if! {
         // fallback for other platforms is to always increase the stack if we're on
         // the root stack. After we increased the stack once, we know the new stack
         // size and don't need this pessimization anymore
+        #[inline(always)]
         unsafe fn guess_os_stack_limit() -> Option<usize> {
```

Comment:
I think we'll want to implement this on wasm32-unknown-unknown, right? It's just the value of the stack pointer if we haven't otherwise specified it I think?

Reply:
Do you mean assuming `--stack-first`?

Reply:
True yeah, I figured all Rust applications would be using that, but do you know of some that aren't? The only downside of leaving this as `None` is that the initial invocation will always grow the stack, but that's perhaps not the end of the world.

Reply:
Done!